### PR TITLE
fixed improper error handling when data is not valid JSON

### PR DESCRIPTION
**Pull request by hackrush01**

### CODE REVIEW

**Comment by jackrobison** [comment]
What does this fix?

File: lbrynet/daemon/auth/server.py [hunk 1]
```diff
@@ -298,12 +298,13 @@ def _render_error(self, failure, request, id_):
...
+            if not failure.check(ComponentsNotStarted, ComponentStartConditionNotMet):
+                log.warning("error processing api request: %s\ntraceback: %s", error.message,
+                            "\n".join(error.traceback))
         else:
             # last resort, just cast it as a string
             error = JSONRPCError(str(failure))
-        if not failure.check(ComponentsNotStarted, ComponentStartConditionNotMet):
-            log.warning("error processing api request: %s\ntraceback: %s", error.message,
-                        "\n".join(error.traceback))
+
```
Comment (eukreign) [inline lbrynet/daemon/auth/server.py]: Above code fails in cases where `failure` is not a `Failure` and thus there is no `.check()` method. Instead it is moved to be inside `if isinstance(failure, Failure)` where `failure.check()` works.

Comment (eukreign) [reply lbrynet/daemon/auth/server.py]: @jackrobison ^

Comment (jackrobison) [reply lbrynet/daemon/auth/server.py]: :+1:

**Pull request merged**